sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -90,26 +91,68 @@ protected virtual void AuthorizeRequest(HttpMessage message)
 
         /// <summary>
         /// Executed in the event a 401 response with a WWW-Authenticate authentication challenge header is received after the initial request.
+        /// The default implementation will attempt to handle Continuous Access Evaluation (CAE) claims challenges.
         /// </summary>
         /// <remarks>Service client libraries may override this to handle service specific authentication challenges.</remarks>
         /// <param name="message">The <see cref="HttpMessage"/> to be authenticated.</param>
         /// <returns>A boolean indicating whether the request was successfully authenticated and should be sent to the transport.</returns>
-        protected virtual ValueTask<bool> AuthorizeRequestOnChallengeAsync(HttpMessage message)
+        protected virtual async ValueTask<bool> AuthorizeRequestOnChallengeAsync(HttpMessage message)
         {
+            if (AuthorizationChallengeParser.IsCaeClaimsChallenge(message.Response) &&
+                TryGetTokenRequestContextForCaeChallenge(message, out var tokenRequestContext))
+            {
+                await AuthenticateAndAuthorizeRequestAsync(message, tokenRequestContext).ConfigureAwait(false);
+                return true;
+            }
+
             return default;
         }
 
         /// <summary>
         /// Executed in the event a 401 response with a WWW-Authenticate authentication challenge header is received after the initial request.
+        /// The default implementation will attempt to handle Continuous Access Evaluation (CAE) claims challenges.
         /// </summary>
         /// <remarks>Service client libraries may override this to handle service specific authentication challenges.</remarks>
         /// <param name="message">The <see cref="HttpMessage"/> to be authenticated.</param>
         /// <returns>A boolean indicating whether the request was successfully authenticated and should be sent to the transport.</returns>
         protected virtual bool AuthorizeRequestOnChallenge(HttpMessage message)
         {
+            if (AuthorizationChallengeParser.IsCaeClaimsChallenge(message.Response) &&
+                TryGetTokenRequestContextForCaeChallenge(message, out var tokenRequestContext))
+            {
+                AuthenticateAndAuthorizeRequest(message, tokenRequestContext);
+                return true;
+            }
             return false;
         }
 
+        internal bool TryGetTokenRequestContextForCaeChallenge(HttpMessage message, out TokenRequestContext tokenRequestContext)
+        {
+            string? decodedClaims = null;
+            string? encodedClaims = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "claims");
```

Comment:
nit: would it make sense to use const strings instead of inline ones here?

Reply:
since they are not used more than once, I don't see much benefit.